target/linux/tcall.c

```diff
@@ -196,6 +198,54 @@ static long _tcall_get_tsd(uint64_t* value)
     return 0;
 }
 
+long myst_tcall_impersonate(long n, long params[6], uid_t uid, gid_t gid)
+{
+    long ret = 0;
+    const long x1 = params[0];
+    const long x2 = params[1];
+    const long x3 = params[2];
+    const long x4 = params[3];
+    const long x5 = params[4];
+    const long x6 = params[5];
+    uid_t existing_uid, existing_euid, existing_savuid;
+    gid_t existing_gid, existing_egid, existing_savgid;
+
```

Comment:
There should be a flag indicating whether the uid and gid are different from the default ones. If not, we can skip these syscalls.

Reply:
This is about the same work both ways and makes the code cleaner to do always I think

Reply:
syscalls are not cheap. I really prefer the optimization.

Reply:
done